arrays/arrays.c

```diff
@@ -83,14 +94,21 @@ char *arr_read(Array *arr, int index) {
 void arr_insert(Array *arr, char *element, int index) {
 
   // Throw an error if the index is greater than the current count
-
+if(arr->count < index){
+  perror("Array only has %s indices.", arr->count);
+}
   // Resize the array if the number of elements is over capacity
-
+if(arr->count == arr->capacity){
+  Array *arr = resize_array(arr);
+}
   // Move every element after the insert index to the right one position
-
+for(int i = arr->count; i > index; i--) {
+  arr->elements[i] = arr->elements[i-1];
+}
   // Copy the element and add it to the array
-
+arr->elements[index] = element;
```

Comment:
You should copy the element using strdup(element) instead of assigning the element directly to arr->elements[index] here.